spec/controllers/state_file/questions/az_qualifying_organization_contributions_controller_spec.rb

```diff
@@ -0,0 +1,208 @@
+require 'rails_helper'
+
+RSpec.describe StateFile::Questions::AzQualifyingOrganizationContributionsController do
+  render_views
+
+  let(:intake) { create :state_file_az_intake }
+
+  before do
+    sign_in intake
+  end
+
+  describe "#index" do
+    it 'should skip when no contributions are added' do
+      get :index
+
+      expect(response).to redirect_to(:new_az_qualifying_organization_contribution)
+    end
+
+    it 'should list contributions' do
+      contribution = create(:az321_contribution, state_file_az_intake: intake)
+
+      get :index
+
+      expect(response).to be_ok
+      expect(response.body).to include(contribution.charity_name)
+    end
+
+    it 'should hide the add another button when three contributions are already created' do
+      create_list(:az321_contribution, 3, state_file_az_intake: intake)
+
+      get :index
+
+      expect(response).to be_ok
+      expect(response).not_to include('Add another contribution')
+    end
+
+    it 'should show the add another button when there are fewer than three contributions' do
+      create(:az321_contribution, state_file_az_intake: intake)
+
+      get :index
+
+      expect(response).to be_ok
+      expect(response.body).to include('Add another contribution')
+    end
+  end
+
+  describe "edit" do
+    let(:contribution) { create(:az321_contribution, state_file_az_intake: intake)}
+
+    it 'should display a form' do
+      get :edit, params: { id: contribution.id }
+
+      expect(response).to be_ok
+      expect(response.body).to include('<form')
```

Comment:
lol, is this a normal test? never seen anything like this before

Reply:
I started writing these as request specs before I realized that we hadn't really begun that migration yet. Since include checks for a string literal I thought I'd just check for the start of a form tag. It _is_ a little wonky though and I can remove the test if you prefer.

Reply:
yeah i think the next block in here (that checks for the contribution data on the page) should cover it